pkg/core/native/native_neo.go

```diff
@@ -932,17 +956,31 @@ func (n *NEO) getAccountState(ic *interop.Context, args []stackitem.Item) stacki
 // ComputeNextBlockValidators returns an actual list of current validators.
 func (n *NEO) ComputeNextBlockValidators(bc interop.Ledger, d *dao.Simple) (keys.PublicKeys, error) {
 	numOfCNs := n.cfg.GetNumOfCNs(bc.BlockHeight() + 1)
-	cache := d.Store.GetRWCache(n.ID).(*NeoCache)
-	if vals := cache.validators; vals != nil && numOfCNs == len(vals) {
+	// Most of the time it should be OK with RO cache, thus try to retrieve
+	// validators without RW cache creation to avoid cached values copying.
+	cache := d.Store.GetROCache(n.ID).(*NeoCache)
+	if vals := cache.validators; numOfCNs == len(vals) {
 		return vals.Copy(), nil
 	}
+	cache = d.Store.GetRWCache(n.ID).(*NeoCache)
+	vals, err := n.computeNextBlockValidatorsFromDAO(bc, d)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compute next block validators: %w", err)
+	}
+	cache.validators = vals
+	return vals, nil
+}
+
+// computeNextBlockValidatorsFromDAO returns an actual list of current validators
+// calculated without cache access.
+func (n *NEO) computeNextBlockValidatorsFromDAO(bc interop.Ledger, d *dao.Simple) (keys.PublicKeys, error) {
```

Comment:
I'm really not sure this needs a `bc`.

Reply:
Fixed, now it accepts current blockchains height instead of `bc`.